lib/read.js

```diff
@@ -55,32 +59,28 @@ class ReadSql extends AdapterRead {
         return !!this.timeField;
     }
 
-    validateOptions(options) {
-        var unknown = _.difference(_.keys(options), ALLOWED_OPTIONS);
-        if (unknown.length > 0) {
-            throw new errors.runtimeError('RT-UNKNOWN-OPTION-ERROR', {
-                proc: 'read-sql',
-                option: unknown[0]
-            });
-        }
+    getDbConnection() {
+        throw new Error('getDBConnection not implemented by adapter');
+    }
 
+    validateOptions(options) {
         if (_.has(options, 'table') && _.has(options, 'raw') ||
             (_.has(options, 'debug') && _.has(options, 'raw'))
         ) {
-            throw new errors.runtimeError('RT-INCOMPATIBLE-OPTION-ERROR', {
+            throw new errors.runtimeError('INCOMPATIBLE-OPTION', {
```

Comment:
Adapters can throw errors through a method on the class, right? I'd suggest changing all of these to this.runtimeError and only importing errors if you really need to.